src/libstd/sys/windows/fs.rs

```diff
@@ -368,18 +364,18 @@ pub fn readlink(p: &Path) -> IoResult<Path> {
     }
     // Specify (sz - 1) because the documentation states that it's the size
     // without the null pointer
-    let ret = fill_utf16_buf_and_decode(|buf, sz| unsafe {
+    let ret = try!(fill_utf16_buf_and_decode(|buf, sz| unsafe {
         GetFinalPathNameByHandleW(handle,
                                   buf as *const u16,
                                   sz - 1,
                                   libc::VOLUME_NAME_DOS)
-    });
+    }));
+    let ret = String::from_utf16(&*ret).unwrap();
     let ret = match ret {
-        Some(ref s) if s.starts_with(r"\\?\") => { // "
+        ref s if s.starts_with(r"\\?\") => { // "
```

Comment:
I'm... not sure. This seems to have been introduced in 40ab198356bbec05b308b2e61ebc6fe9e23ade9d, but I have no idea why!

When the `std::fs` RFC lands I'll try to be sure to see if we can remove this logic.